nixos/modules/services/misc/nix-optimise.nix

```diff
@@ -0,0 +1,49 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.nix.optimise;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    nix.optimise = {
+
+      automatic = mkOption {
```

Comment:
I thought so as well at first, but then it occurred to me the usual semantics of the top-level `enable` option is to act as the identity module if `false`, while this module always defines a new service unit.